File: server/app/Models/Business.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Business extends Model
{
    use HasFactory;

    /**
     * The attributes that are mass assignable.
     *
     * @var list<string>
     */
    protected $fillable = [
        'longitude',
        'latitude',
        'name',
        'address',
        'phone',
        'user_id',
        'type',
    ];

    /**
     * Get the attributes that should be cast.
     *
     * @return array<string, string>
     */
    protected function casts(): array
    {
        return [
            'longitude' => 'decimal:2',
            'latitude' => 'decimal:2',
        ];
    }

    /**
     * Get the user that owns the business.
     */
    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    /**
     * Get the products for the business.
     */
    public function products(): HasMany
    {
        return $this->hasMany(Product::class);
    }

    /**
     * Get the employees for the business.
     */
    public function employees(): HasMany
    {
        return $this->hasMany(Employee::class);
    }

    /**
     * Get the employee roles for the business.
     */
    public function employeeRoles(): HasMany
    {
        return $this->hasMany(EmployeeRole::class, 'business');
    }
}

File: server/Models/Business.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using server.Models;
namespace server.Models
{
    public enum BusinessType
    {
        Retail,
        Service,
        Manufacturing,
        Tourism,
        Other
    }

    public class Business
    {
        [Key]
        public long Id { get; set; }

        [Range(-180, 180)]
        public decimal Longitude { get; set; }

        [Range(-90, 90)]
        public decimal Latitude { get; set; }

        [Required, MaxLength(255)]
        public string Name { get; set; } = null!;

        [Required, MaxLength(255)]
        public string Address { get; set; } = null!;

        [Required, MaxLength(50)]
        public string Phone { get; set; } = null!;

        [Required]
        public BusinessType Type { get; set; }

        // FK
        public long UserId { get; set; }
        public User User { get; set; } = null!;

        // Navigation
        public ICollection<Product> Products { get; set; } = new List<Product>();
        public ICollection<EmployeeRole> EmployeeRoles { get; set; } = new List<EmployeeRole>();

        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
    }

}
